test/runtests.jl

```diff
@@ -96,6 +96,68 @@ end
 @test Interfaces.implements(Tables.DictRowTable, AbstractTable)
 @test Interfaces.implements(Tables.DictColumnTable, AbstractTable)
 
+abstract type Example end
+
+@interface Example T begin
+
+    # required subtyping
+    T <: Example
+
+    # required method definition
+    foo1(arg::Example, arg2::Int)
+
+    # method def w/ return type
+    foo2(arg::Example)::Float64
+
+    # method def w/ interface return type
+    foo3(arg::Example)::Iterable
+
+    # one of many required
+    foo4(arg::Example) || foo5(arg::Example) || foo6(arg::Example)
+
+    # conditional
+    if foo7(T)
+        foo8(arg::Example)
+        foo9(arg::Example)::RT where {RT <: Iterable}
+        foo10(RT)::Int
+    elseif foo11(T)
+        foo12(arg::Example)
+    else
+        foo13(arg::Example) || foo14(arg::Example)
+    end
+
+    @optional foo15(arg::Example)
+
+end
+
+int = Interfaces.interface(Example)
+io = IOBuffer()
+show(io, int)
+@test String(take!(io)) == """
```

Comment:
is there a reason to write to an IOBuffer, rather than using `sprint` or `repr`?
```suggestion
@test repr(int) == """
```
(or `repr("text/plain", int)` if we go with that)